appium/webdriver/appium_service.py

```diff
@@ -64,62 +64,65 @@ class AppiumServiceError(RuntimeError):
     pass
 
 
+T = TypeVar('T', bound='AppiumService')
+
+
 class AppiumService(object):
-    def __init__(self):
-        self._process = None
-        self._cmd = None
+    def __init__(self) -> None:
+        self._process: Optional[sp.Popen] = None
+        self._cmd: Optional[List] = None
 
-    def _get_node(self):
+    def _get_node(self) -> str:
         if not hasattr(self, '_node_executable'):
             self._node_executable = find_executable('node')
         if self._node_executable is None:
             raise AppiumServiceError('NodeJS main executable cannot be found. ' +
                                      'Make sure it is installed and present in PATH')
         return self._node_executable
 
-    def _get_npm(self):
+    def _get_npm(self) -> str:
         if not hasattr(self, '_npm_executable'):
             self._npm_executable = find_executable('npm.cmd' if sys.platform == 'win32' else 'npm')
         if self._npm_executable is None:
             raise AppiumServiceError('Node Package Manager executable cannot be found. ' +
                                      'Make sure it is installed and present in PATH')
         return self._npm_executable
 
-    def _get_main_script(self):
+    def _get_main_script(self) -> str:
         if not hasattr(self, '_main_script'):
             for args in [['root', '-g'], ['root']]:
                 try:
-                    modules_root = subprocess.check_output([self._get_npm()] + args).strip().decode('utf-8')
+                    modules_root = sp.check_output([self._get_npm()] + args).strip().decode('utf-8')
                     if os.path.exists(os.path.join(modules_root, MAIN_SCRIPT_PATH)):
                         self._main_script = os.path.join(modules_root, MAIN_SCRIPT_PATH)
                         break
-                except subprocess.CalledProcessError:
+                except sp.CalledProcessError:
                     continue
             if not hasattr(self, '_main_script'):
                 try:
-                    self._main_script = subprocess.check_output(
+                    self._main_script = str(sp.check_output(
```

Comment:
why it is necessary to cast to str here?

Reply:
It was due to mypy error `appium/webdriver/appium_service.py:103: error: Incompatible types in assignment (expression has type "bytes", variable has type "str")` without cast.
But as you see, it changed current behavior and should be like below in definition.

```
self._main_script: Union[str, bytes] = os.path.join(modules_root, MAIN_SCRIPT_PATH)
```